### PR TITLE
ROU-3052: Sort client actions should return error message and success boolean

### PR DESCRIPTION
**Pull request by gnbm**
This PR adds feedback messages to Sort API methods. Both ClearSort and ColumnSort methods did not return any feedback on whether or not the operation was successful or if any errors happened.

### Test Steps
1. Click in "Column Sort By Product Name":
Expected: Grid should be sorted according to the Product Name column (ascending) and a success message should appear. 

1. Click in "Clear  Sort Function":
Expected: Grid should lose the sorting by Product Name column and a success message should appear. 

1. Click in "Column Sort By Invalid Column":
Expected: Error message should show saying: "Error"

1. Click in "Clear  Sort Invalid":
Expected: Error message should show saying: "Error"

### Checklist
* [ ] tested locally
* [X] documented the code
* [X] clean all warnings and errors of eslint
* [X] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// <autogenerated />
+using System;
+using System.Reflection;
+[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.6.1", FrameworkDisplayName = ".NET Framework 4.6.1")]
```
Comment (BenOsodrac) [inline code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs:4]: @glundgren93 I was not the only one!
@gnbm I think we are not supposed t commit this file

File: code/src/OSFramework/Enum/ErrorCodes.ts [hunk 2]
```diff
@@ -21,6 +21,9 @@ namespace OSFramework.Enum {
         API_FailedApplyRowValidation = 'GRID-API-02005',
         // VIEW
         API_FailedGetViewLayout = 'GRID-API-03001',
-        API_FailedSetViewLayout = 'GRID-API-03002'
+        API_FailedSetViewLayout = 'GRID-API-03002',
+        //SORT
+        API_FailedClear = 'GRID-API-04001',
+        API_FailedColumnSort = 'GRID-API-04002'
```
Comment (BenOsodrac) [inline code/src/OSFramework/Enum/ErrorCodes.ts:27]: We might need to change this after merge, as on my side I also created codes for Selected API using 04...

**Review by BenOsodrac — CHANGES_REQUESTED** [review]

File: code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs (hunk 1 above)
Comment (glundgren93) [reply code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs:4]: Yes, please remove this

Comment (gnbm) [reply code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs:4]: Done!

File: code/src/OSFramework/Enum/ErrorCodes.ts (hunk 2 above)
Comment (BenOsodrac) [reply code/src/OSFramework/Enum/ErrorCodes.ts:27]: Hum I might have missed that part :/

Maybe after the merges we can do another PR to review all the codes

**Review by BenOsodrac — DISMISSED** [review]

**Review by glundgren93 — APPROVED** [review]

**Review by joselrio — DISMISSED** [review]

File: code/src/WijmoProvider/Features/ColumnSort.ts [hunk 3]
```diff
@@ -162,8 +162,25 @@ namespace WijmoProvider.Feature {
...
+                    message: 'Success',
+                    code: OSFramework.Enum.ErrorCodes.GRID_SUCCESS
+                };
+            } catch (error) {
+                return {
+                    isSuccess: false,
+                    message: 'Error',
+                    code: OSFramework.Enum.ErrorCodes.API_FailedClear
+                };
+            }
```
Comment (rugoncalves) [inline code/src/WijmoProvider/Features/ColumnSort.ts:183]: In what situations do exceptions occur?

File: code/src/WijmoProvider/Features/ColumnSort.ts [hunk 4]
```diff
@@ -214,34 +231,58 @@ namespace WijmoProvider.Feature {
...
-            const column = this._grid.getColumn(columnID);
-            const ascending =
-                OSFramework.OSStructure.Sorting[sorting] ===
-                OSFramework.OSStructure.Sorting.Ascending;
+        ): OSFramework.OSStructure.ReturnMessage {
+            try {
+                const column = this._grid.getColumn(columnID);
+                const ascending =
+                    OSFramework.OSStructure.Sorting[sorting] ===
+                    OSFramework.OSStructure.Sorting.Ascending;
```
Comment (rugoncalves) [inline code/src/WijmoProvider/Features/ColumnSort.ts:248]: In what situations does an exception occur?

File: code/styles/Grid.css [hunk 5]
```diff
@@ -289,7 +289,9 @@
     background: var(--datagrid-cell-background);
 }
 
-.wj-cells .wj-cell[aria-readonly='true']:not(.wj-group) > *:not(.wj-cell-maker){
+.wj-cells
+    .wj-cell[aria-readonly='true']:not(.wj-group)
+    > *:not(.wj-cell-maker) {
     pointer-events: none;
```
Comment (rugoncalves) [inline code/styles/Grid.css:295]: Why was this changed?

Comment (gnbm) [reply code/styles/Grid.css:295]: This changed when I ran the lintfix

File: code/src/WijmoProvider/Features/ColumnSort.ts (hunk 3 above)
Comment (gnbm) [reply code/src/WijmoProvider/Features/ColumnSort.ts:183]: When for instance the Grid ID doesn't exist (but will fail first on the method that calls this one). 
In this particular case, I believe that we're doing this to keep consistency and prevent any error thrown from this._grid.provider.collectionView.sortDescriptions.clear();
Don't have a good example for it.

File: code/src/WijmoProvider/Features/ColumnSort.ts (hunk 4 above)
Comment (gnbm) [reply code/src/WijmoProvider/Features/ColumnSort.ts:248]: When we pass for instance an invalid column ID.

File: code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs (hunk 1 above)
Comment (gnbm) [reply code/extension/DataGridDataConvert/Source/NET/obj/Release/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs]: @rugoncalves Do you know why this file was added? Just curious about it since it seems to be part of an OS extension

File: code/src/WijmoProvider/Features/ColumnSort.ts (hunk 3 above)
Comment (rugoncalves) [reply code/src/WijmoProvider/Features/ColumnSort.ts:183]: if we don't have a good example, or any case where a problem occurs, why is this a problem to start with?
I'm a little against using try/catch, if not really needed.

File: code/src/WijmoProvider/Features/ColumnSort.ts (hunk 4 above)
Comment (rugoncalves) [reply code/src/WijmoProvider/Features/ColumnSort.ts:248]: If the column doesn't exist, it will not reach here, right @glundgren93 ?

File: code/styles/Grid.css (hunk 5 above)
Comment (rugoncalves) [reply code/styles/Grid.css:295]: Okay!

File: code/src/WijmoProvider/Features/ColumnSort.ts (hunk 3 above)
Comment (rugoncalves) [reply code/src/WijmoProvider/Features/ColumnSort.ts:183]: Just chatted with Gabriel. Please proceed.

File: code/src/WijmoProvider/Features/ColumnSort.ts (hunk 4 above)
Comment (rugoncalves) [reply code/src/WijmoProvider/Features/ColumnSort.ts:248]: Just chatted with Gabriel. Please proceed.

File: code/src/GridAPI/Sort.ts [hunk 6]
```diff
@@ -8,41 +8,95 @@ namespace GridAPI.Sort {
...
+    export function Clear(gridID: string): string {
         PerformanceAPI.SetMark('Sort.Clear');
 
-        if (!OSFramework.Helper.IsGridReady(gridID)) return;
-        const grid = GridManager.GetGridById(gridID);
+        let returnMessage = {
+            isSuccess: true,
+            message: 'Success',
+            code: OSFramework.Enum.ErrorCodes.GRID_SUCCESS
+        };
```
Comment (rugoncalves) [inline code/src/GridAPI/Sort.ts:20]: Please have an action, to be used cross API to do the set of this default object.

File: code/src/GridAPI/Sort.ts [hunk 7]
```diff
@@ -8,41 +8,95 @@ namespace GridAPI.Sort {
...
 
-        grid.features.sort.clear();
+        if (!OSFramework.Helper.IsGridReady(gridID)) {
+            returnMessage = {
+                isSuccess: false,
+                message: 'Grid not found',
```
Comment (rugoncalves) [inline code/src/GridAPI/Sort.ts:25]: This text should not be hardcoded here. We should have an Enum `ErrorMessages`.

File: code/src/GridAPI/Sort.ts [hunk 8]
```diff
@@ -8,41 +8,95 @@ namespace GridAPI.Sort {
...
+            code: OSFramework.Enum.ErrorCodes.GRID_SUCCESS
+        };
 
-        grid.features.sort.sortColumn(columnID, sorting);
+        if (!OSFramework.Helper.IsGridReady(gridID)) {
+            returnMessage = {
+                isSuccess: false,
+                message: 'Grid not found',
+                code: OSFramework.Enum.ErrorCodes.CFG_GridNotFound
+            };
```
Comment (rugoncalves) [inline code/src/GridAPI/Sort.ts:77]: here you can access to the properties of the type, e.g. `returnMessage.isSuccess = false`.
Also please change the name of the object to something different from "returnMessage" :)

File: code/src/GridAPI/Sort.ts [hunk 9]
```diff
@@ -8,41 +8,95 @@ namespace GridAPI.Sort {
...
+            GridManager.GetGridById(gridID).features.sort.sortColumn(
+                columnID,
+                sorting
+            );
+        } catch (error) {
+            returnMessage = {
+                isSuccess: false,
+                message: 'Error',
+                code: OSFramework.Enum.ErrorCodes.API_FailedClearSort
+            };
```
Comment (rugoncalves) [inline code/src/GridAPI/Sort.ts:90]: The `returnMessage.message` not having the "error" that occured, means that we are completely hiding the error from the developer... Better to do simply: `returnMessage.message = error`.

**Review by rugoncalves — CHANGES_REQUESTED** [review]

File: code/src/GridAPI/Sort.ts (hunk 8 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts:77]: I'm following the same names that Gabriel is using for consistency. I believe we need to review all these changes on a separate task since there are more PRs following this approach.

File: code/src/GridAPI/Sort.ts (hunk 6 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts:20]: I'm following the same names that Gabriel is using for consistency. I believe we need to review all these changes on a separate task since there are more PRs following this approach.

File: code/src/GridAPI/Sort.ts (hunk 7 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts:25]: I already talked about that we @glundgren93 since we have a lot of error messages all over the place

File: code/src/GridAPI/Sort.ts (hunk 8 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts:77]: I'm not good with names..Any suggestion to replace "returnMessage" @rugoncalves ? :)

File: code/src/GridAPI/Sort.ts (hunk 7 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts]: This is done @rugoncalves!

File: code/src/GridAPI/Sort.ts (hunk 8 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts]: This is done @rugoncalves! Only need a better name for the return object

File: code/src/GridAPI/Sort.ts (hunk 9 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts]: This is done @rugoncalves!

**Review by joselrio — DISMISSED** [review]

File: code/src/GridAPI/Sort.ts (hunk 8 above)
Comment (gnbm) [reply code/src/GridAPI/Sort.ts]: This is done @rugoncalves! Renamed to responseObj

**Review by glundgren93 — APPROVED** [review]

**Review by rugoncalves — APPROVED** [review]
